### ISSUE TITLE
Config Option to Ignore Certain Modules

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1904]
**Is your feature request related to a problem? Please describe.**
I'd like a way to specify in the pyright config file to skip a module entirely and treat it as Any. Ideally an equivalent to mypy's skip following-import setting, https://mypy.readthedocs.io/en/stable/running_mypy.html#following-imports.

The motivation is one library I'm using has most of the members I'm using dynamically generated leading to pyright reporting "not a known member of the module". My current solution is to go to every file that imports that module and replace it with,

```
# Nose dynamically generates attributes making it bad choice to type check.
if TYPE_CHECKING:
    t = Any
else:
    from nose import tools as t
```

This works but it'd be nice to not need to duplicate that everywhere. The chance of the library adding type stubs is pretty low as the library hasn't been maintained for years. Admittingly, I could write the stubs myself and may eventually do that for the few files I care about and leave the rest without hints.

**Describe the solution you'd like**
A config option that allows specifying modules as simple regular expressions like (mod1.*, or mod1.submod2.*) and treats any occurrence of those modules as Any.

**Comment by erictraut** [comment]
You can treat all imports from a module as `Any` by simply creating a local stub file for that module that contains a module-level `__getattr__` function that returns `Any`. If you place this stub file in a local directory called `typings`, pyright will automatically use it. (You can choose a directory name other than `typings` by configuring the `stubPath` setting.)

**Comment by erictraut** [comment]
Did my suggested approach solve your problem?

**Comment by erictraut** [comment]
Closing the issue under the assumption that the suggested approach meets your needs.

**Comment by Antyos** [comment]
@erictraut Would it be possible for you to give an example of what a blank stub with `__getattr__` looks like as well as its file structure? 

I have been slamming my face against a wall for the past several hours and I cannot for the life of me figure out how to get Pyright (technically Pylance) to ignore typings for a large 3rd party library that doesn't have stubs. I can open a separate issue with my specific issue if you prefer, but it seemed like it would be a near-duplicate of this one.

**Comment by erictraut** [comment]
Let's say that the module in question is called `coollib`, and it has a submodule called `stuff`. You can create the following files:

```python
# typings/coollib/__init__.pyi
from typing import Any
def __getattr__(name: str) -> Any: ...

# typings/coollib/stuff.pyi
from typing import Any
def __getattr__(name: str) -> Any: ...
```

Now, if your code contains the statement `import coollib` or `import coollib.stuff` or `from coollib import a, b, c` or `from coollib.stuff import d, e, f`, any symbols from those imported modules will have an `Any` type.

**Comment by erictraut** [comment]
I'll also note that pyright, by default, ignores symbols from libraries that don't have stubs. This is controlled by the "useLibraryCodeForTypes" setting, which is off by default for pyright. For pylance, we decided to enable this setting by default because most pylance users are more interested in completion suggestions than in type checking. If you enable type checking, the correctness and completeness of the type information becomes much more important. So if you're using pylance and have enabled type checking, you may want to simply disable "useLibraryCodeForTypes". Pylance will then ignore all libraries that do not have stubs or inlined types.

**Comment by Antyos** [comment]
Thank you so much! (Even more so for the speedy reply!) It's all working now!

I generally (think I) like having the autocompletion from "useLibraryCodeForTypes" enabled. However, Pyright doesn't quite know how to handle some libraries with non-standard import processes. To reduce boilerplate stubs for these libraries and assuming it isn't already a feature, it might be nice to implement something like MyPy's library-specific type overrides ([mypy.readthedocs.io/running_mypy#missing-type-hints-for-third-party-library](https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-type-hints-for-third-party-library)) to `pyrightconfig.json`, though I don't know if that is inline with the objective of Pyright.
